### PR TITLE
Upgraded Vega

### LINKED ISSUE #456: Initial window size for canvas should be at least as big as the plot that appears
I tried to make a heatmap and the plot was bigger than the initial canvas size. I would expect the initial canvas to resize based on how big of a plot it's about to render. This is on macOS High Sierra, here's the repro:

```
x = tc.SArray(range(1,3000000))
y = tc.SArray(range(5,3000004))
heat = tc.visualization.heatmap(x,y)
heat.show()
```

### PR DESCRIPTION
**Pull request by abhishekpratapa**
- Fixes #456
- Upgrades Vega

### CODE REVIEW

File: src/visualization/Turi Create Visualization/src/user_interface/package.json [hunk 1]
```diff
@@ -12,8 +12,8 @@
         "react": "^16.2.0",
         "react-dom": "^16.2.0",
         "react-json-pretty": "^1.7.5",
-        "vega": "^3.2.1",
-        "vega-embed": "^3.2.0",
+        "vega": "4.2.0",
+        "vega-embed": "3.18.2",
         "vega-tooltip": "^0.6.1"
```
Comment (znation) [inline src/visualization/Turi Create Visualization/src/user_interface/package.json]: Do we still need vega-tooltip?

Comment (abhishekpratapa) [reply src/visualization/Turi Create Visualization/src/user_interface/package.json]: Yeah for now with the implementation we have, no issues there, but if we need to change it, it should be a separate PR

Comment (znation) [reply src/visualization/Turi Create Visualization/src/user_interface/package.json]: So we do still need it, but our existing version is still compatible?

Comment (abhishekpratapa) [reply src/visualization/Turi Create Visualization/src/user_interface/package.json]: yeah, we still need it and our existing version is compatible.

**Pull request merged**